client/notification.js

```diff
@@ -130,12 +133,13 @@ _.extend(NotifyClient.prototype, {
    * @param {String} phoneNumber
    * @param {Object} personalisation
    * @param {String} reference
+   * @param {String} smsSenderId
    *
    * @returns {Promise}
    */
-  sendSms: function (templateId, phoneNumber, personalisation, reference) {
+  sendSms: function (templateId, phoneNumber, personalisation, reference, smsSenderId) {
```

Comment:
these functions are getting gross - it'd be a good idea for us to chat to someone with node experience about the best way to pass these in, especially when three things are optional. i wonder if we should require you to pass in an options dict. eg

```js
function (templateId, phoneNumber, options) {
    x = options.personalisation;
    y = options.reference
    z = options.smsSenderId
}
```

Reply:
The idea of passing in an options object was discussed in [this previous PR](https://github.com/alphagov/notifications-node-client/pull/30). The problem was that changing the code to do this will cause a breaking change. But as we'll need to add in an optional letter contact block too, I guess we need to think about what to do here.

Reply:
Agreed to make this change as a separate PR